pkg/elb/cleanup.go

```diff
@@ -199,6 +199,26 @@ func getServiceLoadBalancer(ctx context.Context, ec2API awsapi.EC2, elbAPI Descr
 	return &lb, nil
 }
 
+func getIngressELBName(ctx context.Context, metadataName string, hosts []string) string {
+	// Expected e.g. bf647c9e-default-appingres-350b-1622159649.eu-central-1.elb.amazonaws.com where AWS ALB name is
+	// bf647c9e-default-appingres-350b (cannot be longer than 32 characters).
+	hostNameParts := strings.Split(hosts[0], ".")
+	if len(hostNameParts[0]) == 0 {
+		logger.Debug("%s is ALB Ingress, but probably not provisioned or something other unexpected, skip", metadataName)
+		return ""
+	}
```

Comment:
Thanks @a-hilaly, that's a great point! I tried to revise this part and let `getIngressLoadBalancer` handle it. To have no interruption deletion of `eksctl delete` command, I think it would be better to keep the current business logic to skip the ALB resource clean up [1] if there's unexpected error when getting ELB name.

https://github.com/eksctl-io/eksctl/blob/6e8e73c13cee21b3a833f111e63ade8c03536d04/pkg/elb/cleanup.go#L220-L224

Update: https://github.com/eksctl-io/eksctl/pull/7029/commits/b638eacf9e0b8d75ae174a7a729b59d4cf2a4550#diff-bf8761431813288f0320030b76870e9d4a40dd0e80e1ca725ef262184c498b90R237-R241